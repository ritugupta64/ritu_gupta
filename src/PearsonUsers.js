import React, { Component } from "react";
import UsersGridContainer from "./usersGridContainer";


export class PearsonUsers extends Component {
  constructor(props) {
    super(props);
    this.state = {users: []};
  }

  
  // Make the async call
  componentDidMount(){
    fetch("https://reqres.in/api/users?page=1&per_page=10")
    .then(response=> response.json()).then(data1 =>{this.setState({ users: this.state.users.concat(data1.data) })})
    .catch(error => {
      //console.log("test")
  });
  }

  // Delete Users
  delete = (index, e) => {
      const addresses = this.state.users; 
      const uniqueAddresses = Array.from(new Set(addresses.map(a => a.id)))
                                   .map(id => {
                                        return addresses.find(a => a.id === id)
                                      })
      const del = Object.assign([], uniqueAddresses);
      del.splice(index, 1);
      this.setState({users:del})
  }
  
  render() {
      // Remove dupliacte values from a collection
      const addresses = this.state.users; 
      const uniqueAddresses = Array.from(new Set(addresses.map(a => a.id)))
                                  .map(id => {
                                        return addresses.find(a => a.id === id)
                                      })
                                      
      // Iteration over a collection
      let userGrid = uniqueAddresses.map((item, index) => {
                          return(<UsersGridContainer 
                                                    key = {item.id} 
                                                    firstName = {item.first_name} 
                                                    lastName = {item.last_name} 
                                                    avatar = {item.avatar} delete = {()=>{this.delete(index)}} />)
                                })
  return (
     <React.Fragment>
          {/* ...Users Grid start here */}
          <div className="users"> 
            {this.state.users.length > 0 ? userGrid : <p className="error">All Items have been deleted...</p>}
          </div>
          {/* Users Grid start here... */}
          
    </React.Fragment>
    );
  }
}
