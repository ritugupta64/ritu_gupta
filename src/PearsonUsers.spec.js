import React from "react";
import { shallow } from "enzyme";
import { PearsonUsers } from "./PearsonUsers";

describe('Uesrs List rendering tests', () => {
  const  testState = {
    users: [
      {
        id: 4,
        first_name: "Eve",
        last_name: "Holt",
        avatar:
          "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
      },
      {
        id: 5,
        first_name: "Charles",
        last_name: "Morris",
        avatar:
          "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
      },
      {
        id: 6,
        first_name: "Tracey",
        last_name: "Ramos",
        avatar:
          "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
      }
    ]
  };
  
  it('should behave like tree', () => {
    const wrapper=shallow(
      <PearsonUsers/>
    );
    wrapper.setState(testState);
    expect(wrapper.find('UsersGridContainer').length).toBe(3);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('UsersGridContainer').first().props().delete).toBeDefined();
    expect(wrapper.find('UsersGridContainer').first().props().firstName).toBe('Eve');
    expect(wrapper.find('UsersGridContainer').first().props().lastName).toBe('Holt');
    expect(wrapper.find('UsersGridContainer').first().props().avatar).toBe('https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg');
  });

  it('Test case for delete function', () => {
    const component = shallow(<PearsonUsers />);
    component.setState(testState);
    component.instance().delete(1);
    expect(component.instance().state.users.length).toBe(2);
    component.instance().delete(1);
    expect(component.instance().state.users.length).toBe(1);
  });

  it('should call methodName during componentDidMount', () => {
    const json='hasOwnProperty';
     fetch = jest.fn(() => new Promise(resolve => resolve()));
     const component = shallow(<PearsonUsers />)
     component.instance().componentDidMount();
     expect(fetch).toHaveBeenCalled();
     expect(fetch).toHaveBeenCalledWith('https://reqres.in/api/users?page=1&per_page=10');
     expect(component.instance().state).hasOwnProperty('users');

});
  
});