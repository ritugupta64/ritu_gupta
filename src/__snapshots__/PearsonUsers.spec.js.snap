// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uesrs List rendering tests should behave like tree 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PearsonUsers />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="users"
>
        <UsersGridContainer
                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                delete={[Function]}
                firstName="Eve"
                lastName="Holt"
        />
        <UsersGridContainer
                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                delete={[Function]}
                firstName="Charles"
                lastName="Morris"
        />
        <UsersGridContainer
                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                delete={[Function]}
                firstName="Tracey"
                lastName="Ramos"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <UsersGridContainer
            avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
            delete={[Function]}
            firstName="Eve"
            lastName="Holt"
/>,
          <UsersGridContainer
            avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
            delete={[Function]}
            firstName="Charles"
            lastName="Morris"
/>,
          <UsersGridContainer
            avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
            delete={[Function]}
            firstName="Tracey"
            lastName="Ramos"
/>,
        ],
        "className": "users",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
            "delete": [Function],
            "firstName": "Eve",
            "lastName": "Holt",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
            "delete": [Function],
            "firstName": "Charles",
            "lastName": "Morris",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
            "delete": [Function],
            "firstName": "Tracey",
            "lastName": "Ramos",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="users"
>
          <UsersGridContainer
                    avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                    delete={[Function]}
                    firstName="Eve"
                    lastName="Holt"
          />
          <UsersGridContainer
                    avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                    delete={[Function]}
                    firstName="Charles"
                    lastName="Morris"
          />
          <UsersGridContainer
                    avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                    delete={[Function]}
                    firstName="Tracey"
                    lastName="Ramos"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <UsersGridContainer
              avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
              delete={[Function]}
              firstName="Eve"
              lastName="Holt"
/>,
            <UsersGridContainer
              avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
              delete={[Function]}
              firstName="Charles"
              lastName="Morris"
/>,
            <UsersGridContainer
              avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
              delete={[Function]}
              firstName="Tracey"
              lastName="Ramos"
/>,
          ],
          "className": "users",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
              "delete": [Function],
              "firstName": "Eve",
              "lastName": "Holt",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "function",
            "props": Object {
              "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
              "delete": [Function],
              "firstName": "Charles",
              "lastName": "Morris",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "function",
            "props": Object {
              "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
              "delete": [Function],
              "firstName": "Tracey",
              "lastName": "Ramos",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
